src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,169 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int BASE_ARRAY_LENGTH = 10;
+    private static final double ARRAY_INCREMENT_VALUE = 1.5;
+    private T[] array;
+    private int currentSlot;
+
+    public ArrayList() {
+        array = (T[]) new Object[BASE_ARRAY_LENGTH];
+        currentSlot = -1;
+    }
+
     @Override
     public void add(T value) {
-
+        checkAndIncreaseArrayLength();
+        currentSlot++;
+        array[currentSlot] = value;
```

Comment:
let's just name our `currentSlot` as size and make it start from 0
This will make things easier